types/convert.go

```diff
@@ -654,6 +654,9 @@ func getValidFloatPrefix(sc *stmtctx.StatementContext, s string, isFuncCast bool
 				break
 			}
 			eIdx = i
+		} else if c == '\u0000' {
+			s = s[:validLen]
```

Comment:
This line can be removed, you only need to break from this loop, because line 667 will cut `s`.

Reply:
> This line can be removed, you only need to break from this loop, because line 667 will cut `s`.

Variable `s` need reset, because line 671 compare `validLen` and `len(s)`.
Or I can remove line 658 and modify line 671 to `validLen != len(valid)`.

Reply:
I think both are acceptable then. It would be better if you could make the refactorization.

Reply:
if c < '0' || c > '9'

Reply:
> It will lead to another question. If we modify `validLen != len(valid)` ，when `c <'0' || c> '9'` ， then no exception will be thrown.
> I think it's right now.

sure, then let's just keep it this way.